src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs

```diff
@@ -272,7 +280,7 @@ public void GoForward()
 
 		public bool Navigate(Type sourcePageType, object parameter, NavigationTransitionInfo infoOverride)
 		{
-			var entry = new PageStackEntry(sourcePageType, parameter, infoOverride);
+			var entry = new PageStackEntry(sourcePageType.GetReplacementType(), parameter, infoOverride);
```

Comment:
@nickrandolph I just noticed that in `GetReplacementType`, we're not thread safe.